### PR TITLE
allow mobilecoind-json to accept more than one payload (MCC-1887)

### PR DESCRIPTION
**Pull request by rjwalters**
# Motivation

There are at least three APIs that would benefit from accepting either the complete output of a previous command, or just the relevant data that is required. I am opening this draft to invite discussion over what pattern we want to use for this.

fixes MCC-1887

### CODE REVIEW

File: mobilecoind-json/src/bin/main.rs [hunk 1]
```diff
@@ -478,21 +478,30 @@ fn submit_tx(
...
 
 /// Checks the status of a transfer given a key image and tombstone block
-#[post("/tx/status-as-sender", format = "json", data = "<receipt>")]
+#[post("/tx/status-as-sender", data = "<receipt>")]
 fn check_transfer_status(
     state: rocket::State<State>,
-    receipt: Json<JsonSendPaymentResponse>,
+    receipt: String,
 ) -> Result<Json<JsonStatusResponse>, String> {
+    // allow receipt to be submitted as either JsonSendPaymentResponse or JsonSenderTxReceipt
```
Comment (eranrund) [inline mobilecoind-json/src/bin/main.rs:486]: Why is this needed?
Please use `or_else` instead of having two match statements

**Comment by rjwalters** [comment]
closing to fix

File: mobilecoind-json/src/bin/main.rs (hunk 1 above)
Comment (rjwalters) [reply mobilecoind-json/src/bin/main.rs]: I guess the Sender will always have access to the complete "JsonSendPaymentResponse" so this isn't essential...

It seemed more intuitive to me when scripting `mobilecoind-json` that I would send only the Sender receipt object to this endpoint, since that mirrors what we do on the recipient side.

Comment (rjwalters) [reply mobilecoind-json/src/bin/main.rs]: I was also anticipating that we would use something similar to fix [MCC-1894(]https://mobilecoin.atlassian.net/browse/MCC-1894)

Comment (rjwalters) [reply mobilecoind-json/src/bin/main.rs]: note that both the gRPC and python library pass only the sender receipt to this function
